src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/AnalyzeResultDef.cs

```diff
@@ -7,8 +7,8 @@ namespace Microsoft.DotNet.UpgradeAssistant.Analysis
 {
     public record AnalyzeResultDef
     {
-        public string AnalysisTypeName { get; init; }
+        public string? AnalysisTypeName { get; init; }
 
-        public IAsyncEnumerable<AnalyzeResult> AnalysisResults { get; init; }
+        public IAsyncEnumerable<AnalyzeResult>? AnalysisResults { get; init; }
```

Comment:
Ideally, this would be non-nullable. You could set it to `AsyncEnumerable.Empty<string>` (not sure if that's directly available, but you can trivially add an empty one)